feat: json update promotion step (#3151)
akuity/kargo: docs/docs/35-references/10-promotion-steps.md

|------|------|-------------|
| `commitMessage` | `string` | A description of the change(s) applied by this step. Typically, a subsequent [`git-commit`](#git-commit) step will reference this output and aggregate this commit message fragment with other like it to build a comprehensive commit message that describes all changes. |

### `json-update`

> `json-update` allows you to update the values of specified keys in any JSON file.

Reviewer: Edited for consistency with existing step docs:

```suggestion
`json-update` updates the values of specified keys in any JSON file.
```

#### `json-update` Configuration

> | Field | Type | Required | Description |
> |--------------|------------------|--------------|---------------------------------------------------------------------------------|
> | `path`       | `string`         | Yes          | Path to the JSON file relative to the promotion workspace.                      |
> | `updates`    | `array`          | Yes          | List of key-value pairs to update in the JSON file.                             |
> | `updates.key`| `string`         | Yes          | JSON path (dot-separated) for the key to update.                                |
> | `updates.value`| `any`          | Yes          | Value to set for the specified key. Supports strings, numbers, booleans, arrays, and objects. |

Reviewer: Edited for consistency with existing step docs:

```suggestion
| Name | Type | Required | Description |
|------|------|----------|-------------|
| `path` | `string` | Y | Path to a JSON file. This path is relative to the temporary workspace that Kargo provisions for use by the promotion process. |                      |
| `updates` | `[]object` | Y | The details of changes to be applied to the file. At least one must be specified. |
| `updates[].key`| `string` | Y | JSON path (dot-separated) for the key to update.                                |
| `updates[].value`| `any` | Y | The new value for the key. Typically specified using an expression. Supports strings, numbers, booleans, arrays, and objects. |
```

#### `json-update` Example

```yaml
vars:
- name: gitRepo
  value: https://github.com/example/repo.git
steps:
- uses: git-clone
  config:
    repoURL: ${{ vars.gitRepo }}
    checkout:
    - commit: ${{ commitFrom(vars.gitRepo).ID }}
      path: ./src
    - branch: stage/${{ ctx.stage }}
      create: true
      path: ./out
    - uses: git-clear
      config:
        path: ./out
- uses: json-update
  config:

>     path: configs/settings.json
>     updates:
>     - key: app.version
>       value: "v2.0.0"
>     - key: featureFlags.newFeatureEnabled
>       value: true
>     - key: thresholds.memory.limit
>       value: 512

Reviewer: Something involving an expression would probably be more instructive since this is how the feature is more likely to be used:

```suggestion
    path: configs/settings.json
    updates:
    - key: image.tag
      value: ${{ imageFrom("my/image").Tag }}
```

# Render manifests to ./out, commit, push, etc...
```

#### `json-update` Output

> When `json-update` successfully modifies a JSON file, it generates a `commitMessage` in the following format:

Reviewer: For consistency with existing step docs:

```suggestion
| Name | Type | Description |
|------|------|-------------|
| `commitMessage` | `string` | A description of the change(s) applied by this step. Typically, a subsequent [`git-commit`](#git-commit) step will reference this output and aggregate this commit message fragment with other like it to build a comprehensive commit message that describes all changes. |
```

> When `json-update` successfully modifies a JSON file, it generates a `commitMessage` in the following format:
> 
> ```
> Updated <path>
>   - <key>: <value>
>   - <key>: <value>
> ```

Reviewer: For consistency with existing step docs:

```suggestion
| Name | Type | Description |
|------|------|-------------|
| `commitMessage` | `string` | A description of the change(s) applied by this step. Typically, a subsequent [`git-commit`](#git-commit) step will reference this output and aggregate this commit message fragment with other like it to build a comprehensive commit message that describes all changes. |
```
